keyboards/matrix/falcon_magnet/apc.c

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright 2023 astro
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "apc.h"
+
+#define APC_KEY_STATE_DEFAULT   APC_KEY_OFF
+#define APC_KEY_MIN_DEFAULT     APC_KEY_MAX
+#define APC_KEY_MAX_DEFAULT     APC_KEY_MIN
+#define APC_KEY_LAST_DEFAULT    0
+#define APC_KEY_TRIGGER_DEFAULT 0
+#define APC_KEY_DOWN_DEFAULT    1024
+#define APC_KEY_UP_DEFAULT      0
+
+enum apc_key_state
+{
+    APC_KEY_OFF,
+    APC_KEY_PRESSING,
+    APC_KEY_ON,
+    APC_KEY_RELEASING,
+};
+
+struct apc_key
+{
+    uint32_t state;
+    uint32_t min;
+    uint32_t max;
+    uint32_t last;
+    uint32_t trigger;
+    uint32_t down;
+    uint32_t up;
+    bool on;
+};
+
+static struct apc_key apc_matrix[MATRIX_ROWS][MATRIX_COLS];
+
+#ifdef DYNAMIC_KEYMAP_ENABLE
+#include "dynamic_keymap.h"
+#else
+extern const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS];
+#endif
```

Comment:
This shouldn't be needed, at all.  This code does not need either of these, and should be using `keycode_at_keymap_location` explicitly, as this will handle dynamic keymaps and anything else without the need for preprocessor checks, like this. 

```suggestion
```